x/auth/ante/basic.go

```diff
@@ -200,3 +200,32 @@ func (txh TxTimeoutHeightDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simul
 
 	return next(ctx, tx, simulate)
 }
+
+type TxSigBlockHeightDecorator struct {
+	ak AccountKeeper
+}
+
+func NewTxSigBlockHeightDecorator(ak AccountKeeper) TxSigBlockHeightDecorator {
+	return TxSigBlockHeightDecorator{
+		ak: ak,
+	}
+}
+
+func (txs TxSigBlockHeightDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	if !simulate {
+		params := txs.ak.GetParams(ctx)
+		sbh := tx.GetSigBlockHeight()
+		current := uint64(ctx.BlockHeight())
+		validMin := uint64(0)
+		if current > params.ValidSigBlockPeriod {
+			validMin = current - params.ValidSigBlockPeriod
+		}
+		if sbh > uint64(ctx.BlockHeight()) || sbh < validMin {
```

Comment:
Does it exist any replay protection logic within `ValidSigBlockPeriod`?

Reply:
We will have to restrict the deletion of an account that signed ever within this `ValidSigBlockPeriod`. Although this implementation is not necessary because there is no account deletion now, it should be remembered when implementing account deletion in the future.